### PR TITLE
Add ColorManager to Vectors

### PR DESCRIPTION
**Pull request by kevinyamauchi**
# Description
This pull request integrates `ColorManager` into the `Vectors` layer. This is a non-breaking change. We will update the public API in a future PR (see https://github.com/napari/napari/issues/1727#issuecomment-788917315).

## Type of change
- [ ] Bug-fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References
This is a follow up to #2204 and discussed in #1727

# How has this been tested?
- [x] ll tests pass with my change

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/guides/stable/translations.html).

### CODE REVIEW

File: napari/layers/vectors/vectors.py [hunk 1]
```diff
@@ -185,27 +181,37 @@ def __init__(
             properties=Event,
         )
 
-        self.visible = False
+        # self.visible = False
```
Comment (goanpeca) [inline napari/layers/vectors/vectors.py]: Uncoment/remove?

**Review by goanpeca — APPROVED** [review]
LGTM, just one comment that was left in code maybe?

File: napari/layers/vectors/vectors.py (hunk 1 above)
Comment (kevinyamauchi) [reply napari/layers/vectors/vectors.py]: Oops, sorry about that. Fixed!

**Comment by kevinyamauchi** [comment]
Thanks for the review, @goanpeca ! I've deleted the line that was commented out.

**Review by sofroniewn — APPROVED** [review]
LGTM! Nice job @kevinyamauchi!!

**Review by goanpeca — APPROVED** [review]
🥳

**Pull request merged**